includes/class-wc-stripe-database-cache-prefetch.php

```diff
@@ -0,0 +1,220 @@
+			WC_Stripe_Logger::debug( 'Enqueued cache prefetch', $logging_context );
+		}
+	}
+
+	/**
+	 * Check if a prefetch is already queued up.
+	 *
+	 * @param string $key The unprefixed cache key to check.
+	 * @return bool True if a prefetch is queued up, false otherwise.
+	 */
+	private function is_prefetch_queued( string $key ): bool {
+		if ( ! isset( self::PREFETCH_CONFIG[ $key ] ) ) {
+			return false;
+		}
+
+		$prefetch_option_key = $this->get_prefetch_option_name( $key );
+
+		$prefetch_option = get_option( $prefetch_option_key, false );
+		// We use ctype_digit() and the (string) cast to ensure we handle the option value being returned as a string.
+		if ( ! ctype_digit( (string) $prefetch_option ) ) {
+			return false;
+		}
+
+		$now             = time();
+		$prefetch_window = self::PREFETCH_CONFIG[ $key ];
+
+		if ( $prefetch_option >= ( $now - $prefetch_window ) ) {
+			// If the prefetch entry expires in the future, or falls within the prefetch window for the key, we should consider the item live and queued.
+			// We use a prefetch window buffer to account for latency on the prefetch processing and to make sure we don't prefetch more than once during the prefetch window.
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get the name of the prefetch tracking option for a given cache key.
+	 *
+	 * @param string $key The unprefixed cache key to get the option name for.
+	 * @return string The name of the prefetch option.
+	 */
+	private function get_prefetch_option_name( string $key ): string {
+		return self::PREFETCH_OPTION_PREFIX . $key;
+	}
+
+	/**
+	 * Handle the prefetch action. We are generally expecting this to be queued up by Action Scheduler using
+	 * the action from {@see ASYNC_PREFETCH_ACTION}.
+	 *
+	 * @param string $key The unprefixed cache key to prefetch.
+	 * @return void
+	 */
+	public function handle_prefetch_action( $key ): void {
+		if ( ! is_string( $key ) || empty( $key ) ) {
+			WC_Stripe_Logger::warning(
+				'Invalid cache prefetch key',
+				[
+					'cache_key' => $key,
+					'reason'    => 'invalid_key',
+				]
+			);
+			return;
+		}
+
+		if ( ! $this->should_prefetch_cache_key( $key ) ) {
+			WC_Stripe_Logger::warning(
+				'Invalid cache prefetch key',
+				[
+					'cache_key' => $key,
+					'reason'    => 'unsupported_cache_key',
+				]
+			);
+			return;
+		}
+
+		$this->prefetch_cache_key( $key );
+
+		// Regardless of whether the prefetch was successful or not, we should remove the prefetch tracking option.
+		delete_option( $this->get_prefetch_option_name( $key ) );
+	}
+
+	/**
+	 * Helper method to implement prefetch/repopulation for supported cache entries.
+	 *
+	 * @param string $key The unprefixed cache key to prefetch.
+	 * @return bool|null True if the prefetch was successful, false if the prefetch failed, or null if the prefetch was not attempted.
+	 */
+	protected function prefetch_cache_key( string $key ): ?bool {
+		$prefetched = null;
+
+		switch ( $key ) {
```

Comment:
Is there a reason to have a switch with a single key? Maybe it is a pattern I am not aware of

Reply:
I set this up as a way to make future additions easier, as adding another supported prefetch key should only require adding an entry to `PREFETCH_CONFIG` and implementing the case in this `switch` statement. That optimisation may be very premature, especially considering that the only other prefetch possibility that I can think of is the account data.